Dockerfile

```diff
@@ -44,9 +44,9 @@ RUN dnf install -y python3 python3-requests && \
     dnf upgrade -y && \
     dnf --setopt=install_weak_deps=False install -y $(cat /tmp/main-packages-list.txt) && \
     dnf clean all && \
-    rm -rf /var/cache/{yum,dnf}/*
+    rm -rf /var/cache/{yum,dnf}/* && \
+    mkdir -p /tftpboot
 
-RUN mkdir -p /tftpboot
```

Comment:
IIUC this command isn't even needed, because COPY will create the directory for you.
A lot more layers could be saved by combining this and the next few lines:

```
COPY --from=builder /tmp/ipxe/src/bin/undionly.kpxe /tmp/ipxe/src/bin-x86_64-efi/snponly.efi /tmp/ipxe/src/bin-x86_64-efi/ipxe.efi /tftpboot/
```

Reply:
correct, although it would be more prone to invalidating cache, so it's a deal beetween reducing build time or pull/push time.
I guess we probably prefer the latter?

Reply:
I don't understand how that would make it more prone to invalidating the cache tbh.

Reply:
because you have more files in one COPY command, if just one of them changes, the entire layer is invalidated in the cache

Reply:
Right, but even with multiple COPY commands if the first one changes then the cache of the subsequent layers is discarded anyway. I don't see any a priori reason to think that the files in the later commands are more likely to change than those in the earlier commands, and even if there were then I'd expect it to have a negligible impact on build time.